File: docs/Drasil/src/java/Truss/OutputFormat.java
package Truss;

/** \file OutputFormat.java
    \author Ting-Yu Wu
    \brief Provides the function for writing outputs
*/
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;

public class OutputFormat {
    
    /** \brief Writes the output values to output.txt
        \param inParams structure holding the input values
        \param F_vect_AC force between joint A and C (N)
        \param F_vect_AD force between joint A and D (N)
        \param F_vect_BC force between joint B and C (N)
        \param F_vect_BD force between joint B and D (N)
    */
    public static void write_output(InputParameters inParams, double F_vect_AC, double F_vect_AD, double F_vect_BC, double F_vect_BD) throws IOException {
        PrintWriter outfile;
        outfile = new PrintWriter(new FileWriter(new File("log.txt"), true));
        outfile.println("function write_output called with inputs: {");
        outfile.print("  inParams = ");
        outfile.print("Instance of InputParameters object");
        outfile.println(", ");
        outfile.print("  F_vect_AC = ");
        outfile.print(F_vect_AC);
        outfile.println(", ");
        outfile.print("  F_vect_AD = ");
        outfile.print(F_vect_AD);
        outfile.println(", ");
        outfile.print("  F_vect_BC = ");
        outfile.print(F_vect_BC);
        outfile.println(", ");
        outfile.print("  F_vect_BD = ");
        outfile.println(F_vect_BD);
        outfile.println("  }");
        outfile.close();
        
        PrintWriter outputfile;
        outputfile = new PrintWriter(new FileWriter(new File("output.txt"), false));
        outputfile.print("F_vect_Ax = ");
        outputfile.println(inParams.F_vect_Ax);
        outputfile.print("F_vect_Ay = ");
        outputfile.println(inParams.F_vect_Ay);
        outputfile.print("F_vect_By = ");
        outputfile.println(inParams.F_vect_By);
        outputfile.print("F_vect_AC = ");
        outputfile.println(F_vect_AC);
        outputfile.print("F_vect_AD = ");
        outputfile.println(F_vect_AD);
        outputfile.print("F_vect_BC = ");
        outputfile.println(F_vect_BC);
        outputfile.print("F_vect_BD = ");
        outputfile.println(F_vect_BD);
        outputfile.print("F_vect_CD = ");
        outputfile.println(inParams.F_vect_CD);
        outputfile.close();
    }
}

File: docs/Drasil/src/swift/OutputFormat.swift
/** OutputFormat.swift
    Provides the function for writing outputs
    - Authors: Ting-Yu Wu
*/
import Foundation

/** Writes the output values to output.txt
    - Parameter inParams: structure holding the input values
    - Parameter F_vect_AC: force between joint A and C (N)
    - Parameter F_vect_AD: force between joint A and D (N)
    - Parameter F_vect_BC: force between joint B and C (N)
    - Parameter F_vect_BD: force between joint B and D (N)
*/
func write_output(_ inParams: inout InputParameters, _ F_vect_AC: Double, _ F_vect_AD: Double, _ F_vect_BC: Double, _ F_vect_BD: Double) throws -> Void {
    var outfile: FileHandle
    do {
        outfile = try FileHandle(forWritingTo: FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!.appendingPathComponent("log.txt"))
        try outfile.seekToEnd()
    } catch {
        throw "Error opening file."
    }
    do {
        try outfile.write(contentsOf: Data("function write_output called with inputs: {".utf8))
        try outfile.write(contentsOf: Data("\n".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outfile.write(contentsOf: Data("  inParams = ".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outfile.write(contentsOf: Data("Instance of InputParameters object".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outfile.write(contentsOf: Data(", ".utf8))
        try outfile.write(contentsOf: Data("\n".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outfile.write(contentsOf: Data("  F_vect_AC = ".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outfile.write(contentsOf: Data(String(F_vect_AC).utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outfile.write(contentsOf: Data(", ".utf8))
        try outfile.write(contentsOf: Data("\n".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outfile.write(contentsOf: Data("  F_vect_AD = ".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outfile.write(contentsOf: Data(String(F_vect_AD).utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outfile.write(contentsOf: Data(", ".utf8))
        try outfile.write(contentsOf: Data("\n".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outfile.write(contentsOf: Data("  F_vect_BC = ".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outfile.write(contentsOf: Data(String(F_vect_BC).utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outfile.write(contentsOf: Data(", ".utf8))
        try outfile.write(contentsOf: Data("\n".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outfile.write(contentsOf: Data("  F_vect_BD = ".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outfile.write(contentsOf: Data(String(F_vect_BD).utf8))
        try outfile.write(contentsOf: Data("\n".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outfile.write(contentsOf: Data("  }".utf8))
        try outfile.write(contentsOf: Data("\n".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outfile.close()
    } catch {
        throw "Error closing file."
    }
    
    var outputfile: FileHandle
    do {
        outputfile = try FileHandle(forWritingTo: FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!.appendingPathComponent("output.txt"))
    } catch {
        throw "Error opening file."
    }
    do {
        try outputfile.write(contentsOf: Data("F_vect_Ax = ".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.write(contentsOf: Data(String(inParams.F_vect_Ax).utf8))
        try outputfile.write(contentsOf: Data("\n".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.write(contentsOf: Data("F_vect_Ay = ".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.write(contentsOf: Data(String(inParams.F_vect_Ay).utf8))
        try outputfile.write(contentsOf: Data("\n".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.write(contentsOf: Data("F_vect_By = ".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.write(contentsOf: Data(String(inParams.F_vect_By).utf8))
        try outputfile.write(contentsOf: Data("\n".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.write(contentsOf: Data("F_vect_AC = ".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.write(contentsOf: Data(String(F_vect_AC).utf8))
        try outputfile.write(contentsOf: Data("\n".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.write(contentsOf: Data("F_vect_AD = ".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.write(contentsOf: Data(String(F_vect_AD).utf8))
        try outputfile.write(contentsOf: Data("\n".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.write(contentsOf: Data("F_vect_BC = ".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.write(contentsOf: Data(String(F_vect_BC).utf8))
        try outputfile.write(contentsOf: Data("\n".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.write(contentsOf: Data("F_vect_BD = ".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.write(contentsOf: Data(String(F_vect_BD).utf8))
        try outputfile.write(contentsOf: Data("\n".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.write(contentsOf: Data("F_vect_CD = ".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.write(contentsOf: Data(String(inParams.F_vect_CD).utf8))
        try outputfile.write(contentsOf: Data("\n".utf8))
    } catch {
        throw "Error printing to file."
    }
    do {
        try outputfile.close()
    } catch {
        throw "Error closing file."
    }
}
